libres/lib/enkf/time_map.cpp

```diff
@@ -373,30 +352,45 @@ bool time_map_try_update(time_map_type *map, int step, time_t time) {
     bool updateOK;
     time_map_assert_writable(map);
     pthread_rwlock_wrlock(&map->rw_lock);
-    { updateOK = time_map_update__(map, step, time); }
+    {
+        updateOK = time_map_update__(map, step, time,
+                                     time_t_vector_safe_iget(map->map, step));
+    }
     pthread_rwlock_unlock(&map->rw_lock);
     return updateOK;
 }
 
 bool time_map_summary_update(time_map_type *map, const ecl_sum_type *ecl_sum) {
-    bool updateOK = time_map_try_summary_update(map, ecl_sum);
+    time_map_assert_writable(map);
+    bool updateOK = true;
+    pthread_rwlock_wrlock(&map->rw_lock);
+    {
 
-    if (!updateOK) {
-        time_map_summary_log_mismatch(map, ecl_sum);
-    }
+        int first_step = ecl_sum_get_first_report_step(ecl_sum);
+        int last_step = ecl_sum_get_last_report_step(ecl_sum);
+        int step;
 
-    return updateOK;
-}
+        for (step = first_step; step <= last_step; step++) {
```

Comment:
It is possible to ride yourself of the `updateOK` statefull loop here with `std::all_of` if we make a `ecl_sum_report_step_iterator` which should iterate over those report step indecies that the `ecl_sum` "has"
```cpp
auto it = ecl_sum_report_step_iterator(ecl_sum);
bool updateOK = std::all_of(it.begin(), it.end(), [&map, &sim_time] (auto step) { 
    return time_map_update__(map, step, time_t_vector_safe_iget(map->map, step));
});
```

Reply:
Yes, sounds like a good idea, however I was planning to change the return value of `time_map_update__` from `bool` to `std::string` as it should report an error message if it is unable to update.